### DISCUSSION TITLE
Cannot create new codespace from pull request branch

### DISCUSSION THREAD

**Discussion by JRoy** [community/community#4014] [Codespaces] [answered]
Currently if you want to resolve a complex merge conflict in a pull request from a repository for which you have push access to in a  codespace, that is currently impossible (even though you are able to push to the pull request branch) and requires a local clone.

It would would great if you could create a codespace from pull requests you have push access to rather than just being able to create codespaces for repositories you are invited to. 

The UX of this could either possibly be a button on the pull request sidebar to create a codespace from the pull request or show a pull request section in the branch selector in the normal codespace creation page.

**Comment by 2percentsilk** [comment, marked as answer]
Hi @JRoy, if you open the page for a Pull Request on a repo you have contributor access to, there should be a Open with... button on the top right corner and if the repository allows codespaces to be created from it, you should see the following
![image](https://user-images.githubusercontent.com/3174849/120671765-8ef35a80-c446-11eb-8c81-508c474e029c.png)
